src/Validation/Validator.php

```diff
@@ -202,7 +211,7 @@ public function errors(array $data, bool $newRecord = true): array
 
             $canBeEmpty = $this->_canBeEmpty($field, $context);
 
-            $flags = static::EMPTY_ALL;
+            $flags = static::EMPTY_NULL;
```

Comment:
I can't shake the feeling that this behavior change is going to come back and bite us. Should we instead be requiring people to use `allowEmptyFor('field', Validator::EMPTY_NULL)` to get the new behavior?

Reply:
My understanding from this change was it moved the default of EMPTY_ALL from `errors()` to the `allowEmptyFor()` call.  I'm not sure implicit null == EMPTY_NULL is the cleanest, but I don't know how many other options follow that.

Reply:
> My understanding from this change was it moved the default of EMPTY_ALL from errors() to the allowEmptyFor() call.

Yup.

@markstory All test cases still pass. The only difference the change causes is that error message shown. For e.g. earlier if you didn't use `notEmptyString()` for a field as just use `minLength()` then submitting a field with empty string you got the generic `This field cannot be empty` message. But not since empty string is not considered empty by default the validator is run and you get the error message specific to `minLength()` validator. I don't think this will cause any real world problems.

Reply:
If users do have issues which can swap back the constants in a bugfix release.